plugin/trino-exasol/src/main/java/io/trino/plugin/exasol/ExasolClient.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    @Override
+    public void createSchema(ConnectorSession session, String schemaName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support creating schemas");
+    }
+
+    @Override
+    public void dropSchema(ConnectorSession session, String schemaName, boolean cascade)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support dropping schemas");
+    }
+
+    @Override
+    public void renameSchema(ConnectorSession session, String schemaName, String newSchemaName)
+    {
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming schemas");
+    }
+
+    @Override
+    protected void renameTable(ConnectorSession session, Connection connection, String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
+    {
+        if (!remoteSchemaName.equals(newRemoteSchemaName)) {
+            throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables across schemas");
+        }
+        throw new TrinoException(NOT_SUPPORTED, "This connector does not support renaming tables");
+    }
+
+    @Override
+    protected boolean isSupportedJoinCondition(ConnectorSession session, JdbcJoinCondition joinCondition)
+    {
+        // Deactivated because test 'testJoinPushdown()' requires write access which is not implemented for Exasol
+        return false;
+    }
+
+    @Override
+    public Optional<JdbcExpression> implementAggregation(ConnectorSession session, AggregateFunction aggregate, Map<String, ColumnHandle> assignments)
+    {
+        // Deactivated because test 'testCaseSensitiveAggregationPushdown()' requires write access which is not implemented for Exasol
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<ColumnMapping> toColumnMapping(ConnectorSession session, Connection connection, JdbcTypeHandle typeHandle)
+    {
+        Optional<ColumnMapping> mapping = getForcedMappingToVarchar(typeHandle);
+        if (mapping.isPresent()) {
+            return mapping;
+        }
+        switch (typeHandle.jdbcType()) {
+            case Types.BOOLEAN:
+                return Optional.of(booleanColumnMapping());
+            case Types.SMALLINT:
+                return Optional.of(smallintColumnMapping());
+            case Types.INTEGER:
+                return Optional.of(integerColumnMapping());
+            case Types.BIGINT:
+                return Optional.of(bigintColumnMapping());
+            case Types.DOUBLE:
+                return Optional.of(doubleColumnMapping());
+            case Types.DECIMAL:
+                int decimalDigits = typeHandle.requiredDecimalDigits();
+                int columnSize = typeHandle.requiredColumnSize();
+                return Optional.of(decimalColumnMapping(createDecimalType(columnSize, decimalDigits)));
```

Comment:
According to https://docs.exasol.com/db/latest/sql_references/data_types/datatypedetails.htm, Exasol DECIMAL is limited to precision of 36 while Trino goes up to 38.

Please add tests where a Trino query compares a decimal column in exasol with a DECIMAL(38, s) literal in Trino to make sure behaviour is correct and adjust the implementation accordingly.

Also the decimal related tests in the type-mapping tests class are missing higher precision tests, please add those. If they fail then add explicit assertions on the failures to make sure the connector fails as expected instead of returning wrong results or behaving in unexpected manner.

Reply:
Can you please give me an example for similar tests in another plugin?